core/src/edu/cornell/gdiac/main/LevelModel.java

```diff
@@ -657,20 +668,31 @@ public void makeTransparentObject(SimpleObstacle obstacle, AssetDirectory direct
 
 
   public void createFrames(SimpleObstacle obstacle, AssetDirectory directory, String key, int rows,
-      int cols, int frame) {
-    Animator.create(obstacle, directory, key, rows, cols, frame);
+      int cols, int frame, AnimationType type) {
+    Animator.create(obstacle, directory, key, rows, cols, frame, type);
   }
 
-  public void nextFrame(SimpleObstacle obstacle) {
-    Animator.getTextureRegion(obstacle);
+  public void nextFrame(SimpleObstacle obstacle, AnimationType type) {
+    Animator.getTextureRegion(obstacle, type);
   }
 
   public void animateObjects() {
-    if (avatar.getPlayerFrames() != null) {
-      nextFrame(avatar);
+    // player animate with type
+    AnimationType type = IDLE;
+    if (Math.abs(avatar.getVX()) >= MIN_WALK_SPEED) {
+      type = WALK;
     }
+    if (avatar.getIsJumping() || !avatar.getJumpingAnimationStatus()) {
+      // animate jump if 1) player initiates a jump 2) jumping animation hasn't finished
+      type = JUMP;
+    }
+    if (avatar.getPlayerFrames(type) != null) {
+      nextFrame(avatar, type);
+    }
```

Comment:
The `PlayerModel` should be responsible for providing the information about its animation type, not the `LevelModel`